### ISSUE TITLE
New compiler infrastructure does not handle redirects correctly

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno#8154]
```ts
// main.ts
import { serve } from "https://deno.land/std/http/server.ts";
```

#### 1.4.6

```
❯❯❯ rm -rf $DENO_DIR
❯❯❯ deno1.4.6 run -A --reload main.ts
Download https://deno.land/std/http/server.ts
Warning Implicitly using latest version (0.75.0) for https://deno.land/std/http/server.ts
(snip)
Check file:///mnt/f9/Projects/github.com/denoland/deno/main.ts
```

#### 1.5.0

```
❯❯❯ rm -rf $DENO_DIR
❯❯❯ deno1.5.0 run -A --reload main.ts
Download https://deno.land/std/http/server.ts
Warning Implicitly using latest version (0.75.0) for https://deno.land/std/http/server.ts
(snip)
Check file:///mnt/f9/Projects/github.com/denoland/deno/main.ts
error: TS2322 [ERROR]: Type 'import("https://deno.land/std@0.75.0/http/server.ts").ServerRequest | null' is not assignable to type 'import("https://deno.land/std/http/server.ts").ServerRequest | null'.
  Type 'import("https://deno.land/std@0.75.0/http/server.ts").ServerRequest' is not assignable to type 'import("https://deno.land/std/http/server.ts").ServerRequest'.
    Types have separate declarations of a private property '_contentLength'.
        request = await readRequest(conn, reader);
        ~~~~~~~
    at https://deno.land/std/http/server.ts:147:9
```

cc @kitsonk 

--- 

For anyone looking at this issue now, we are planning to do a 1.5.1 hot fix release tomorrow with a bug fix.

**Comment by kitsonk** [comment]
Simpler repo:

```
> deno cache https://deno.land/x/std/http/server.ts
Check https://deno.land/x/std/http/server.ts
error: TS2322 [ERROR]: Type 'import("https://deno.land/std@0.75.0/http/server.ts").ServerRequest | null' is not assignable to type 'import("https://deno.land/x/std/http/server.ts").ServerRequest | null'.
  Type 'import("https://deno.land/std@0.75.0/http/server.ts").ServerRequest' is not assignable to type 'import("https://deno.land/x/std/http/server.ts").ServerRequest'.
    Types have separate declarations of a private property '_contentLength'.
        request = await readRequest(conn, reader);
        ~~~~~~~
    at https://deno.land/x/std/http/server.ts:147:9
```

**Comment by kitsonk** [comment]
I've got this fixed, but I want to rant a little bit... `std/http/server.ts` has some really bad and arbitrary circular dependencies... `std/http/_io.ts` imports `std/http/server.ts` which imports `std/http/_io.ts`.  With the redirect bug, `std/http/server.ts` was essentially two different modules in TypeScript when type checking, but the lack of these circular dependencies coupled with redirects is why the bug went undiscovered.  So I am glad we uncovered it, but we need to try to avoid circular dependencies in the `std` library, especially non-sensical arbitrary ones.
